### PR TITLE
Update Electron and associated dependencies

### PR DESCRIPTION
**Pull request by belcherj**
### Fix
This updates the Electron dependencies to the latest. In doing so we need to
have a minimum of Node 12 which is required by Electron 7. We also need to move
the Windows build from AppVeyor to CircleCi.

### Test
Test the Electron builds. This is a multiple point update so we should expect something to break.

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated with:

- Update Electron and associated dependencies

### CODE REVIEW

**Comment by nsakaimbo** [comment]
Tried to take a closer look at the Windows code-signing failure, which I'm running into locally.

```
error   P R I 1 9 1 :   0 x 8 0 0 7 0 0 7 e   -   A p p x   m a n i f e s t   n o t   f o u n d   o r   i s   i n v a l i d .   P l e a s e   e n s u r e   w e l l - f o r m e d   m a n i f e s t   f i l e   i s   p r e s e n t .   O r   s p e c i f y   a n   i n d e x   n a m e   w i t h   / i n   s w i t c h
```
===================

- The Appx manifest _is_ present in the location passed to `makepri.exe`
- What's odd is the error is occurring for me on both `develop` and the `upgrade` branch, which is perplexing. Perhaps the borked state is being cached somehow?
- Inspecting pre- and post- upgrade xml manifests yields the following, but not sure if the differences are material or not.

```
BEFORE: Publisher='CN="Automattic, Inc.", O="Automattic, Inc.", L=San Francisco, S=California, C=US'
AFTER: Publisher='CN="Automattic, Inc.",O="Automattic, Inc.",L=San Francisco,ST=California,C=US'
```
```
BEFORE:  <uap:DefaultTile Wide310x150Logo="assets\Wide310x150Logo.png" />
        
      
    </Application>
  </Applications>
</Package>
      </uap:VisualElements>
      
AFTER: Wide310x150Logo="assets\Wide310x150Logo.png" />
        
      </uap:VisualElements>
      
    </Application>
  </Applications>
</Package>
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -108,6 +106,51 @@ jobs:
...
+      - run:
+          name: Install node version
+          command: |
+            $NODE_VERSION = Get-Content .\calypso\.nvmrc
+            nvm install $NODE_VERSION
+            nvm use $NODE_VERSION
+      - run:
+          name: Npm install
+          command: |
+            # The release of nodejs currently being used by Calypso includes a buggy version
```
Comment (nsakaimbo) [inline .circleci/config.yml]: Can probably update these comments so they don't refer to Calypso. 

It might also be worth verifying whether or not you need this fix at all (i.e. if the version of Node you're building with doesn't need this patch, then you might as well leave it out).

File: .circleci/config.yml [hunk 2]
```diff
@@ -108,6 +106,51 @@ jobs:
...
+            npm config set node_gyp "$(Get-Location)\node_modules\node-gyp\bin\node-gyp.js"
+            npm ci
+      - run:
+          name: Build windows
+          command: |
+            # Workaround for Sign Tool "private key filter" bug in Circle's Windows image.
+            # Ref: https://travis-ci.community/t/codesigning-on-windows/
+            # 
+            # Fix: Import .p12 into the local certificate store. Sign Tool will use 
+            # package.json's `certificateSubjectName` to find the imported cert.
```
Comment (nsakaimbo) [inline .circleci/config.yml]: You'll also need to add `certificateSubjectName` key to package.json (you can see how this was done in wp-desktop).

([ref](https://github.com/Automattic/wp-desktop/blob/d45d0d947b1d2d67b38580f9744197869f110e94/package.json#L150))

Comment (belcherj) [reply .circleci/config.yml]: https://github.com/Automattic/simplenote-electron/pull/1771/commits/fc7a2b225fcdc527e8755e2520e66db2d266e425

File: .circleci/config.yml (hunk 1 above)
Comment (belcherj) [reply .circleci/config.yml]: https://github.com/Automattic/simplenote-electron/pull/1771/commits/f2055d78a0cd4796947cb0d113176be749467e02

File: package.json [hunk 3]
```diff
@@ -66,7 +66,7 @@
     "@types/redux": "3.6.0",
     "@types/redux-localstorage": "1.0.8",
     "@types/rimraf": "2.0.3",
-    "@types/webdriverio": "4.13.3",
+    "@types/webdriverio": "5.0.0",
```
Comment (dmsnell) [inline package.json]: according to the `package.json` in `spectron`, version 9 of `spectron` (necessary for `electron@7`) still uses `webdriverio@4` so we will want to undo this change.

the good news is that starting with `spectron@10` it pulls in the `webdriver` and `@types/webdriverio` packages for us 👍

**Comment by dmsnell** [comment]
## Testing

 - Started by removing `node_modules`, setting `node` to `12.14.1`
 - Ran `npm test` and all tests passed
 - Ran `npm run dev` and it failed with `Unable to find electron's version number, either install it or specify an explicit version`

Are you able to build this locally?

File: package.json (hunk 3 above)
Comment (dmsnell) [reply package.json]: reverted version bump in 65c5288

**Comment by belcherj** [comment]
Windows build

## Testing
 
 - [ ] Logout
 - [ ] Login with wrong password fails
 - [ ] Login with correct password
 - [ ] Create new note appears in other device
 - [ ] Changes to new note sync to other device
 - [ ] Changes to new note sync _from_ other device (after debounce delay)
 - [ ] New tag immediately syncs to other device
 - [ ] New tag immediately syncs _from_ other device
 - [ ] Removed tag immediately syncs to other device
 - [ ] Removed tag immediately syncs _from_ other device
 - [ ] Note publishes with link
 - [ ] Note unpublishes
 - [ ] Note publish change syncs _from_ other device (visible with dialog open)
 - [ ] Markdown setting syncs _from_ other device
 - [ ] Preview mode disappears/reappears when receiving remote changes to markdown setting
 - [ ] Note pinning syncs immediately from either direction
 - [ ] Note pinning works regardless if clicking in list view or from note info
 - [ ] Viewing history on one device leaves note unchanged on other device
 - [ ] Restoring history immediately syncs note from both directions
 - [ ] Restoring history revisions updates pinned status
 - [ ] Restoring history revisions updates markdown status
 - [ ] Restoring history revisions updates publish status
 - [ ] Can view trashed notes by clicking on `Trash`
 - [ ] Can delete note forever from trash screen
 - [ ] Can restore note from trash screen
 - [ ] Can trash note
 - [ ] Can preview markdown with 👁 button
 - [ ] Can flip to edit mode with 👁 button
 - [ ] Markdown rendered in note list when markdown enabled
 - [ ] Markdown syntax unrendered in note list when markdown disabled
 - [ ] Can filter by tag when clicking on tag in tag drawer
 - [ ] Can search by keyword with tag selected
 - [ ] Searching in the search field highlights matches in note list
 - [ ] Searching in the search field highlights matches in the note editor
 - [ ] Searching in the search field highlights matches in the note editor during preview
 - [ ] Clearing the search field immediately updates filtered notes
 - [ ] Clicking on different tags or `All Notes` or `Trash` immediately updates filtered notes
 - [ ] Can search by keyword, filtering debounced
 - [ ] Tag auto-completes appear when typing in search field
 - [ ] Typing `tag:` does not result a list of all tags in the autocompleter
 - [ ] Typing `tag:` and something else, like `tag:te` results in autocomplete results starting with that something else, e.g. `test`
 - [ ] Search field updates with results of `tag:test` format search string
 - [ ] Can toggle sidebar
 - [ ] Syncs when introducing sequential surrogate pairs sharing the same high surrogate, e.g. `🅰🅱` to `🅰🅰🅱`
 - [ ] When disabling network connectivity, changes start counting in unsync'ed changes counter
 - [ ] When going back online changes sync and counter resets to `All chagnes synced`
 - [ ] Can change analytics sharing setting
 - [ ] Changing `Note Display` mode immediately updates and reflects in note list
 - [ ] With sidebar disabled, toggling `Line Length` between `Narrow` and `Full` removes and adds border around note content appropriately and immediately.
 - [ ] Changing `Sort Type` mode immediately updates and reflects in note list
 - [ ] Toggling `Sort Order` immediately updates and reflects in note list for each sort type
 - [ ] For each sort type the pinned notes appear first in the note list
 - [ ] Changing `Theme` immediately updates app for desired color scheme
 - [ ] Using the `Insert checklist` item from the format menu inserts a checklist
 - [ ] "Undo" undoes the last edit
 - [ ] Typing ` - [ ] ` creates a checked checklist item
 - [ ] Typing ` - [ ] ` created an unchecked checklist item
 - [ ] Typing ` - ` creates a list
 - [ ] Typing _tab_ in a list item underneath another list item indents item
 - [ ] Typing _shift tab_ in the same spot unindents the list
 - [ ] Changing `-` to `+` changes the list item bullet, also for `*` and `•` (`\u2022`)
 - [ ] All list bullet types render to markdown lists
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + <kbd>p</kbd> toggles preview mode
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + (<kbd>k</kbd> or <kbd>up</kbd>) selects next note above current note, stops at top of list
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + (<kbd>j</kbd> or <kbd>down</kbd>) selects next note below current note, stops at bottom of list
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>t</kbd> toggles tab list
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>n</kbd> creates new note
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + <kbd>n</kbd> _when in small screen mode_ navigates to note list

**Comment by belcherj** [comment]
Linux build

## Testing
 
 - [ ] Logout
 - [ ] Login with wrong password fails
 - [ ] Login with correct password
 - [ ] Create new note appears in other device
 - [ ] Changes to new note sync to other device
 - [ ] Changes to new note sync _from_ other device (after debounce delay)
 - [ ] New tag immediately syncs to other device
 - [ ] New tag immediately syncs _from_ other device
 - [ ] Removed tag immediately syncs to other device
 - [ ] Removed tag immediately syncs _from_ other device
 - [ ] Note publishes with link
 - [ ] Note unpublishes
 - [ ] Note publish change syncs _from_ other device (visible with dialog open)
 - [ ] Markdown setting syncs _from_ other device
 - [ ] Preview mode disappears/reappears when receiving remote changes to markdown setting
 - [ ] Note pinning syncs immediately from either direction
 - [ ] Note pinning works regardless if clicking in list view or from note info
 - [ ] Viewing history on one device leaves note unchanged on other device
 - [ ] Restoring history immediately syncs note from both directions
 - [ ] Restoring history revisions updates pinned status
 - [ ] Restoring history revisions updates markdown status
 - [ ] Restoring history revisions updates publish status
 - [ ] Can view trashed notes by clicking on `Trash`
 - [ ] Can delete note forever from trash screen
 - [ ] Can restore note from trash screen
 - [ ] Can trash note
 - [ ] Can preview markdown with 👁 button
 - [ ] Can flip to edit mode with 👁 button
 - [ ] Markdown rendered in note list when markdown enabled
 - [ ] Markdown syntax unrendered in note list when markdown disabled
 - [ ] Can filter by tag when clicking on tag in tag drawer
 - [ ] Can search by keyword with tag selected
 - [ ] Searching in the search field highlights matches in note list
 - [ ] Searching in the search field highlights matches in the note editor
 - [ ] Searching in the search field highlights matches in the note editor during preview
 - [ ] Clearing the search field immediately updates filtered notes
 - [ ] Clicking on different tags or `All Notes` or `Trash` immediately updates filtered notes
 - [ ] Can search by keyword, filtering debounced
 - [ ] Tag auto-completes appear when typing in search field
 - [ ] Typing `tag:` does not result a list of all tags in the autocompleter
 - [ ] Typing `tag:` and something else, like `tag:te` results in autocomplete results starting with that something else, e.g. `test`
 - [ ] Search field updates with results of `tag:test` format search string
 - [ ] Can toggle sidebar
 - [ ] Syncs when introducing sequential surrogate pairs sharing the same high surrogate, e.g. `🅰🅱` to `🅰🅰🅱`
 - [ ] When disabling network connectivity, changes start counting in unsync'ed changes counter
 - [ ] When going back online changes sync and counter resets to `All chagnes synced`
 - [ ] Can change analytics sharing setting
 - [ ] Changing `Note Display` mode immediately updates and reflects in note list
 - [ ] With sidebar disabled, toggling `Line Length` between `Narrow` and `Full` removes and adds border around note content appropriately and immediately.
 - [ ] Changing `Sort Type` mode immediately updates and reflects in note list
 - [ ] Toggling `Sort Order` immediately updates and reflects in note list for each sort type
 - [ ] For each sort type the pinned notes appear first in the note list
 - [ ] Changing `Theme` immediately updates app for desired color scheme
 - [ ] Using the `Insert checklist` item from the format menu inserts a checklist
 - [ ] "Undo" undoes the last edit
 - [ ] Typing ` - [ ] ` creates a checked checklist item
 - [ ] Typing ` - [ ] ` created an unchecked checklist item
 - [ ] Typing ` - ` creates a list
 - [ ] Typing _tab_ in a list item underneath another list item indents item
 - [ ] Typing _shift tab_ in the same spot unindents the list
 - [ ] Changing `-` to `+` changes the list item bullet, also for `*` and `•` (`\u2022`)
 - [ ] All list bullet types render to markdown lists
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + <kbd>p</kbd> toggles preview mode
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + (<kbd>k</kbd> or <kbd>up</kbd>) selects next note above current note, stops at top of list
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + (<kbd>j</kbd> or <kbd>down</kbd>) selects next note below current note, stops at bottom of list
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>t</kbd> toggles tab list
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>n</kbd> creates new note
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + <kbd>n</kbd> _when in small screen mode_ navigates to note list

**Comment by belcherj** [comment]
- Moving from to preview right after making changes on the Windows app losses those changes.
- The system menu does not work on windows.

**Comment by dmsnell** [comment]
> Moving from to preview right after making changes on the Windows app losses those changes. Broken in previous version, did update after taking some time.

This should be fixed in `develop` and a rebase would catch it. It was introduced in the selected note refactor not in this branch

**Comment by belcherj** [comment]
Both checklists complete, no new bugs.  I think it is ready to ship!!!

File: lib/dialogs/import/index.tsx [hunk 4]
```diff
@@ -8,12 +8,7 @@ import TransitionDelayEnter from '../../components/transition-delay-enter';
 import Spinner from '../../components/spinner';
 
 const SourceImporter = React.lazy(() =>
-  import(
-    /*
-    webpackChunkName: 'source-importer',
-    webpackPrefetch: true,
```
Comment (dmsnell) [inline lib/dialogs/import/index.tsx]: is there an explanation for why we're removing this? I didn't see any `webpack` version changes or implications from the rest of the PR why this would need to go.

**Comment by dmsnell** [comment]
## Testing
 
 - [x] Logout
 - [x] Login with wrong password fails
 - [x] Login with correct password
 - [x] Create new note appears in other device
 - [x] Changes to new note sync to other device
 - [x] Changes to new note sync _from_ other device (after debounce delay)
 - [x] New tag immediately syncs to other device
 - [x] New tag immediately syncs _from_ other device
 - [x] Removed tag immediately syncs to other device
 - [x] Removed tag immediately syncs _from_ other device
 - [x] Note publishes with link
 - [x] Note unpublishes
 - [x] Note publish change syncs _from_ other device (visible with dialog open)
 - [x] Markdown setting syncs _from_ other device
 - [x] Preview mode disappears/reappears when receiving remote changes to markdown setting
 - [x] Note pinning syncs immediately from either direction
 - [x] Note pinning works regardless if clicking in list view or from note info
 - [x] Viewing history on one device leaves note unchanged on other device
 - [x] Restoring history immediately syncs note from both directions ⭕️ (on the web app I saw the revision immediately appear after restoring in Electron but then it also immediately undid the restore. after the restore I have both windows open with different versions of the note. the web app has two new revisions and reloading doesn't change anything. the electron app doesn't show those changes but after a reload it does. I recreated the sequence and made another edit in the web version and although that change appeared in the note list, the note editor didn't receive the update. switching notes and coming back resolved it)
 - [ ] Restoring history revisions updates pinned status ⭕️ (also in `develop`)
 - [ ] Restoring history revisions updates markdown status ⭕️ (also in `develop`)
 - [ ] Restoring history revisions updates publish status ⭕️ (also in `develop`)
 - [x] Can view trashed notes by clicking on `Trash`
 - [x] Can delete note forever from trash screen
 - [x] Can restore note from trash screen
 - [x] Can trash note
 - [x] Can preview markdown with 👁 button
 - [x] Can flip to edit mode with 👁 button
 - [x] Markdown rendered in note list when markdown enabled
 - [x] Markdown syntax unrendered in note list when markdown disabled
 - [x] Can filter by tag when clicking on tag in tag drawer
 - [x] Can search by keyword with tag selected
 - [x] Searching in the search field highlights matches in note list
 - [x] Searching in the search field highlights matches in the note editor
 - [ ] Searching in the search field highlights matches in the note editor during preview ⭕️ (also in `develop`)
 - [x] Clearing the search field immediately updates filtered notes
 - [x] Clicking on different tags or `All Notes` or `Trash` immediately updates filtered notes
 - [x] Can search by keyword, filtering debounced
 - [x] Tag auto-completes appear when typing in search field
 - [x] Typing `tag:` does not result a list of all tags in the autocompleter
 - [x] Typing `tag:` and something else, like `tag:te` results in autocomplete results starting with that something else, e.g. `test`
 - [x] Search field updates with results of `tag:test` format search string
 - [x] Can toggle sidebar
 - [x] Syncs when introducing sequential surrogate pairs sharing the same high surrogate, e.g. `🅰🅱` to `🅰🅰🅱`
 - [x] When disabling network connectivity, changes start counting in unsync'ed changes counter
 - [x] When going back online changes sync and counter resets to `All chagnes synced`
 - [x] Can change analytics sharing setting
 - [x] Changing `Note Display` mode immediately updates and reflects in note list
 - [x] With sidebar disabled, toggling `Line Length` between `Narrow` and `Full` removes and adds border around note content appropriately and immediately.
 - [x] Changing `Sort Type` mode immediately updates and reflects in note list
 - [x] Toggling `Sort Order` immediately updates and reflects in note list for each sort type
 - [x] For each sort type the pinned notes appear first in the note list
 - [x] Changing `Theme` immediately updates app for desired color scheme
 - [x] Using the `Insert checklist` item from the format menu inserts a checklist
 - [x] "Undo" undoes the last edit
 - [x] Typing ` - [ ] ` creates a checked checklist item
 - [x] Typing ` - [ ] ` created an unchecked checklist item
 - [x] Typing ` - ` creates a list
 - [x] Typing _tab_ in a list item underneath another list item indents item
 - [x] Typing _shift tab_ in the same spot unindents the list
 - [x] Changing `-` to `+` changes the list item bullet, also for `*` and `•` (`\u2022`)
 - [ ] All list bullet types render to markdown lists ⭕️ (`\u2022` doesn't render as a list in rendered Markdown, also in `develop`)
 - [x] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + <kbd>p</kbd> toggles preview mode
 - [x] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + (<kbd>k</kbd> or <kbd>up</kbd>) selects next note above current note, stops at top of list
 - [x] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + (<kbd>j</kbd> or <kbd>down</kbd>) selects next note below current note, stops at bottom of list
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>t</kbd> toggles tab list ⭕️ (also in `develop`)
 - [x] <kbd>CmdOrCtrl</kbd> + <kbd>n</kbd> creates new note
 - [ ] <kbd>CmdOrCtrl</kbd> + <kbd>shift</kbd> + <kbd>n</kbd> _when in small screen mode_ navigates to note list

File: lib/dialogs/import/index.tsx (hunk 4 above)
Comment (dmsnell) [reply lib/dialogs/import/index.tsx]: From Slack:

```
init.ts:204 Unable to load preload script: ~/simplenote-electron/desktop/preload.js
```

I was unable to reproduce this error but I have seen similar ones before. It seems like this is probably fine to leave in but it's also the only invocation of it in the repository. The directive will preload the script during CPU idle time, meaning that imports start a hair earlier.

All that being said, while I think it's going to be better _either_ leaving this in _or_ taking it out in a separate PR with an explanation for the removal, it seems like such a small enough issue and already that it's an outlier that I don't think we need to block this PR to undo the change.

**Comment by dmsnell** [comment]
> Restoring history immediately syncs note from both directions

With further testing this appears related to tags more than content
